File: app/src/main/java/io/ermdev/classify/data/local/teacher/TeacherDao.java
package io.ermdev.classify.data.local.teacher;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

/**
 * Created by erafaelmanuel on 12/16/2017.
 */

@Dao
public interface TeacherDao {

    @Query("SELECT * FROM tblteacher WHERE id=:teacherId")
    Teacher getById(long teacherId);

    @Query("SELECT * FROM tblteacher")
    List<Teacher> getAll();

    @Insert
    void insert(Teacher teacher);

    @Insert
    void insertAll(Teacher... teachers);

    @Update
    void update(Teacher teachers);

    @Update
    void updateAll(Teacher... teachers);

    @Delete
    void delete(Teacher teacher);

    @Delete
    void deleteAll(Teacher... teachers);
}

File: app/src/main/java/io/ermdev/classify/data/local/dao/TeacherDao.kt
package io.ermdev.classify.data.local.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import io.ermdev.classify.data.entity.Teacher

/**
 * Created by erafaelmanuel on 1/3/2018.
 */
@Dao
interface TeacherDao {

    @Query("SELECT * FROM tblteacher WHERE id=:teacherId")
    fun findById(teacherId: Long): Teacher

    @Query("SELECT * FROM tblteacher")
    fun findAll(): Teacher

    @Insert
    fun save(teacher: Teacher)
}